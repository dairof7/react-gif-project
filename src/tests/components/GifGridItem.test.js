const { shallow } = require("enzyme")
const { GifGridItem } = require("../../components/GifGridItem")
import React from 'react'


describe('Pruebas en <GifGridItem />', () => {
    
    const title = 'titulo'
    const url = 'https://localhost/hola.jpg'
    const wrapper = shallow ( <GifGridItem title={ title } url={ url }/> )
    
    test('should show correctly the component', () => {
        expect ( wrapper ).toMatchSnapshot();
    })

    test('should show a <p> with the title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    })

    test('should show the image same tnah url', () => {
        const img = wrapper.find('img');
        //imprime el html de la img completo, en un string
        //console.log(img.html())
        //con props puedo analizar mas facilmente las propiedades
        
        expect( img.props().src ).toBe(url);
        expect( img.props().alt ).toBe(title);
    })
    
    test('should have animate_backInDown className', () => {
        const div = wrapper.find('div')
        const class_Name = div.props().className
        expect( class_Name.includes( 'animate__backInDown' ) ).toBe(true);
        
    })
    

})
