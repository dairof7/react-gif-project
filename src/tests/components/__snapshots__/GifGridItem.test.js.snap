// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> should show correctly the component 1`] = `
<div
  className="card animate__animated animate__backInDown"
>
  <img
    alt="titulo"
    src="https://localhost/hola.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
